inception-external-search-solr/src/main/java/de/tudarmstadt/ukp/inception/externalsearch/solr/SolrSearchProvider.java

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright 2020
+ * Ubiquitous Knowledge Processing (UKP) Lab
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.externalsearch.solr;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.conn.HttpHostConnectException;
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.BaseHttpSolrClient;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrDocumentList;
+import org.apache.solr.common.SolrException;
+
+import de.tudarmstadt.ukp.clarin.webanno.text.TextFormatSupport;
+import de.tudarmstadt.ukp.inception.externalsearch.ExternalSearchHighlight;
+import de.tudarmstadt.ukp.inception.externalsearch.ExternalSearchProvider;
+import de.tudarmstadt.ukp.inception.externalsearch.ExternalSearchResult;
+import de.tudarmstadt.ukp.inception.externalsearch.model.DocumentRepository;
+import de.tudarmstadt.ukp.inception.externalsearch.solr.traits.SolrSearchProviderTraits;
+
+
+public class SolrSearchProvider
+        implements ExternalSearchProvider<SolrSearchProviderTraits>
+{
+
+    private static final String DOC_ID_KEY = "id";
+    private static final String DOC_NAME_KEY = "name";
+    private static final String DOC_TITLE_KEY = "title";
+
+    private static final String DOC_URI_KEY = "uri";
+    private static final String DOC_LANGUAGE_KEY = "language";
+    private static final String DOC_TIMESTAMP_KEY = "timestamp";
+
+    /**
+     * Search documents in a Solr repository
+     * @param aRepository Solr repository
+     * @param aTraits Param for client connection
+     * @param aQuery The query that we retrieve from the search bar
+     * @return A list of results that Inception can read and display
+     * @throws SolrException Connection exception and request exception
+     */
+
+    public List<ExternalSearchResult> executeQuery(DocumentRepository aRepository,
```

Comment:
Please format new/modified code using the DKPro Core Formatting template for Eclipse linked [here](https://dkpro.github.io/contributing/) under code style - please do not entirely re-format existing files - only sections you changed. Note that the formatter rules apply only to Java files. For XML files, please indent using two spaces - avoid reformatting entire XML files (in particular comments).

See also: 
* https://inception-project.github.io/releases/0.16.0/docs/developer-guide.html#_setting_up_checkstyle_and_formatting
* https://inception-project.github.io/releases/0.16.0/docs/developer-guide.html#_checkstyle_and_formatting_216